observation_portal/proposals/admin.py

```diff
@@ -111,6 +114,46 @@ def get_actions(self, request):
             del actions['delete_selected']
         return actions
 
+    @admin.action(description='Make proposals Public')
+    def makepublic_selected(self, request, queryset):
+        public = queryset.update(public=True)
+        self.message_user(request, 'Successfully made {} proposal(s) public'.format(public))
+
+    @admin.action(description='Rollover time allocation for selected proposals')
+    def rollover_selected(self, request, queryset):
+        now = timezone.now()
+        currentsemester = Semester.objects.filter(start__lte=now, end__gte=now).first()
+        nextsemester = Semester.objects.filter(start__gte=currentsemester.end).order_by('start').first()
+        if 'apply' in request.POST:
+            for obj in queryset:
+                allocations = obj.timeallocation_set.filter(semester=currentsemester)
+                if obj.timeallocation_set.filter(semester=nextsemester):
```

Comment:
you should explicitly call .exists() on the queryset here

Reply:
Done